extensions/sql-database-projects/src/dialogs/updateProjectFromDatabaseDialog.ts

```diff
@@ -0,0 +1,605 @@
+		this.dialog.registerCloseValidator(async () => {
+			return this.validate();
+		});
+	}
+
+	public async openDialog(): Promise<void> {
+		let connection = await azdata.connection.getCurrentConnection();
+		if (connection) {
+			this.connectionId = connection.connectionId;
+		}
+
+		this.initializeDialog();
+
+		this.dialog.okButton.label = constants.updateProjectDialogOkButtonText;
+		this.dialog.okButton.enabled = false;
+		this.toDispose.push(this.dialog.okButton.onClick(async () => await this.handleUpdateButtonClick()));
+
+		this.dialog.cancelButton.label = constants.cancelButtonText;
+
+		getAzdataApi()!.window.openDialog(this.dialog);
+		await this.initDialogPromise;
+
+		this.tryEnableUpdateButton();
+	}
+
+	private dispose(): void {
+		this.toDispose.forEach(disposable => disposable.dispose());
+	}
+
+	private initializeDialog(): void {
+		this.initializeUpdateProjectFromDatabaseTab();
+		this.dialog.content = [this.updateProjectFromDatabaseTab];
+	}
+
+	private initializeUpdateProjectFromDatabaseTab(): void {
+		this.updateProjectFromDatabaseTab.registerContent(async view => {
+
+			const connectionRow = this.createServerRow(view);
+			const databaseRow = this.createDatabaseRow(view);
+			const sourceDatabaseFormSection = view.modelBuilder.flexContainer().withLayout({ flexFlow: 'column' }).component();
+			sourceDatabaseFormSection.addItems([connectionRow, databaseRow]);
+
+			const projectLocationRow = this.createProjectLocationRow(view);
+			const folderStructureRow = this.createFolderStructureRow(view);
+			const targetProjectFormSection = view.modelBuilder.flexContainer().withLayout({ flexFlow: 'column' }).component();
+			targetProjectFormSection.addItems([projectLocationRow, folderStructureRow]);
+
+			const actionRow = this.createActionRow(view);
+			const actionFormSection = view.modelBuilder.flexContainer().withLayout({ flexFlow: 'column' }).component();
+			actionFormSection.addItems([actionRow]);
+
+			this.formBuilder = <azdata.FormBuilder>view.modelBuilder.formContainer()
+				.withFormItems([
+					{
+						title: constants.sourceDatabase,
+						components: [
+							{
+								component: sourceDatabaseFormSection,
+							}
+						]
+					},
+					{
+						title: constants.targetProject,
+						components: [
+							{
+								component: targetProjectFormSection,
+							}
+						]
+					},
+					{
+						title: constants.updateAction,
+						components: [
+							{
+								component: actionFormSection,
+							}
+						]
+					}
+				], {
+					horizontal: false,
+					titleFontSize: cssStyles.titleFontSize
+				})
+				.withLayout({
+					width: '100%',
+					padding: '10px 10px 0 20px'
+				});
+
+			let formModel = this.formBuilder.component();
+			await view.initializeModel(formModel);
+			this.connectionButton?.focus();
+			this.initDialogComplete?.resolve();
+		});
+	}
+
+	private createServerRow(view: azdata.ModelView): azdata.FlexContainer {
```

Comment:
a lot of these components are the same as the ones used in other dialogs. It would be good to be able to reuse instead of duplicating code, maybe have a base dialog class.